[Pytorch Mobile DebugInfo Serialization] Save debug handles for all instructions.

kimishpatel opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #57481 [Pytorch Delegated Backend] Save function name in debug info
* #57441 [Pytorch Delegated Backend] Save operator name and function name in
* #57156 [Pytorch Delegated Backend] Add python binding for generate_debug_handles
* #55462 [Pytorch Backend delegation] Add api for backend lowering to query debug
* #56562 [Inlined Callstack Fix] Fix inlined callstack for blocks of the node.
* **#55252 [Pytorch Mobile DebugInfo Serialization] Save debug handles for all instructions.**
* #55062 [Pytorch, Mobile] Serialize inlined callstack pointer with debug handle.
* #54284 [PyTorch, Mobile] Serialization format change for source range

Summary:
Earlier for bytecode serialization we were saving debug handles only for OPs and not all
instructions. This PR makes changes to add that for all instructions.

Test Plan:
python test/mobile/test_lite_script_module.py TestLiteScriptModule

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27542502](https://our.internmc.facebook.com/intern/diff/D27542502)

kimishpatel committed:
[Moible DebugInfo Serialization] Save debug handles for all instructions.

Summary:
Earlier for bytecode serialization we were saving debug handles only for OPs and not all
instructions. This PR makes changes to add that for all instructions.

Test Plan:
python test/mobile/test_lite_script_module.py TestLiteScriptModule

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

kimishpatel committed:
Update on "[Moible DebugInfo Serialization] Save debug handles for all instructions."


Summary:
Earlier for bytecode serialization we were saving debug handles only for OPs and not all
instructions. This PR makes changes to add that for all instructions.

Test Plan:
python test/mobile/test_lite_script_module.py TestLiteScriptModule

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27542502](https://our.internmc.facebook.com/intern/diff/D27542502)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Moible DebugInfo Serialization] Save debug handles for all instructions."


Summary:
Earlier for bytecode serialization we were saving debug handles only for OPs and not all
instructions. This PR makes changes to add that for all instructions.

Test Plan:
python test/mobile/test_lite_script_module.py TestLiteScriptModule

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27542502](https://our.internmc.facebook.com/intern/diff/D27542502)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Moible DebugInfo Serialization] Save debug handles for all instructions."


Summary:
Earlier for bytecode serialization we were saving debug handles only for OPs and not all
instructions. This PR makes changes to add that for all instructions.

Test Plan:
python test/mobile/test_lite_script_module.py TestLiteScriptModule

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27542502](https://our.internmc.facebook.com/intern/diff/D27542502)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Moible DebugInfo Serialization] Save debug handles for all instructions."


Summary:
Earlier for bytecode serialization we were saving debug handles only for OPs and not all
instructions. This PR makes changes to add that for all instructions.

Test Plan:
python test/mobile/test_lite_script_module.py TestLiteScriptModule

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27542502](https://our.internmc.facebook.com/intern/diff/D27542502)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Moible DebugInfo Serialization] Save debug handles for all instructions."


Summary:
Earlier for bytecode serialization we were saving debug handles only for OPs and not all
instructions. This PR makes changes to add that for all instructions.

Test Plan:
python test/mobile/test_lite_script_module.py TestLiteScriptModule

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27542502](https://our.internmc.facebook.com/intern/diff/D27542502)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Moible DebugInfo Serialization] Save debug handles for all instructions."


Summary:
Earlier for bytecode serialization we were saving debug handles only for OPs and not all
instructions. This PR makes changes to add that for all instructions.

Test Plan:
python test/mobile/test_lite_script_module.py TestLiteScriptModule

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27542502](https://our.internmc.facebook.com/intern/diff/D27542502)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Moible DebugInfo Serialization] Save debug handles for all instructions."


Summary:
Earlier for bytecode serialization we were saving debug handles only for OPs and not all
instructions. This PR makes changes to add that for all instructions.

Test Plan:
python test/mobile/test_lite_script_module.py TestLiteScriptModule

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27542502](https://our.internmc.facebook.com/intern/diff/D27542502)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Moible DebugInfo Serialization] Save debug handles for all instructions."


Summary:
Earlier for bytecode serialization we were saving debug handles only for OPs and not all
instructions. This PR makes changes to add that for all instructions.

Test Plan:
python test/mobile/test_lite_script_module.py TestLiteScriptModule

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27542502](https://our.internmc.facebook.com/intern/diff/D27542502)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Moible DebugInfo Serialization] Save debug handles for all instructions."


Summary:
Earlier for bytecode serialization we were saving debug handles only for OPs and not all
instructions. This PR makes changes to add that for all instructions.

Test Plan:
python test/mobile/test_lite_script_module.py TestLiteScriptModule

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27542502](https://our.internmc.facebook.com/intern/diff/D27542502)

[ghstack-poisoned]

dreiss commented on test/mobile/test_lite_script_module.py:
```diff
@@ -379,8 +379,13 @@ def forward(self, x, y, w):
             loaded(torch.rand(3, 4), torch.rand(3, 4), torch.rand(30, 40))
+        except RuntimeError as e:
+            error_message = f"{e}"
+        self.assertTrue('test_lite_script_module.py\", line {}'.format(lineno + 3) in error_message)
```
This will fail with a confusing undefined variable error if the module doesn't raise a RuntimeError.

dreiss commented on test/mobile/test_lite_script_module.py:
```diff
@@ -398,6 +403,34 @@ def forward(self, x, w):
+                super(FooTest5, self).__init__()
+                self.val = val
+
+            @torch.jit.script_method
```
I thought these were deprecated and you're supposed to call `torch.jit.script` on the module instance.

raziel replied:
That's right.
The only annotations that are still ok to use are @torch.jit.ignore or @torch.jit.unused.

kimishpatel committed:
Update on "[Moible DebugInfo Serialization] Save debug handles for all instructions."


Summary:
Earlier for bytecode serialization we were saving debug handles only for OPs and not all
instructions. This PR makes changes to add that for all instructions.

Test Plan:
python test/mobile/test_lite_script_module.py TestLiteScriptModule

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27542502](https://our.internmc.facebook.com/intern/diff/D27542502)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Moible DebugInfo Serialization] Save debug handles for all instructions."


Summary:
Earlier for bytecode serialization we were saving debug handles only for OPs and not all
instructions. This PR makes changes to add that for all instructions.

Test Plan:
python test/mobile/test_lite_script_module.py TestLiteScriptModule

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27542502](https://our.internmc.facebook.com/intern/diff/D27542502)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Moible DebugInfo Serialization] Save debug handles for all instructions."


Summary:
Earlier for bytecode serialization we were saving debug handles only for OPs and not all
instructions. This PR makes changes to add that for all instructions.

Test Plan:
python test/mobile/test_lite_script_module.py TestLiteScriptModule

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27542502](https://our.internmc.facebook.com/intern/diff/D27542502)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Pytorch Mobile DebugInfo Serialization] Save debug handles for all instructions."


Summary:
Earlier for bytecode serialization we were saving debug handles only for OPs and not all
instructions. This PR makes changes to add that for all instructions.

Test Plan:
python test/mobile/test_lite_script_module.py TestLiteScriptModule

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27542502](https://our.internmc.facebook.com/intern/diff/D27542502)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Pytorch Mobile DebugInfo Serialization] Save debug handles for all instructions."


Summary:
Earlier for bytecode serialization we were saving debug handles only for OPs and not all
instructions. This PR makes changes to add that for all instructions.

Test Plan:
python test/mobile/test_lite_script_module.py TestLiteScriptModule

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27542502](https://our.internmc.facebook.com/intern/diff/D27542502)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Pytorch Mobile DebugInfo Serialization] Save debug handles for all instructions."


Summary:
Earlier for bytecode serialization we were saving debug handles only for OPs and not all
instructions. This PR makes changes to add that for all instructions.

Test Plan:
python test/mobile/test_lite_script_module.py TestLiteScriptModule

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27542502](https://our.internmc.facebook.com/intern/diff/D27542502)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Pytorch Mobile DebugInfo Serialization] Save debug handles for all instructions."


Summary:
Earlier for bytecode serialization we were saving debug handles only for OPs and not all
instructions. This PR makes changes to add that for all instructions.

Test Plan:
python test/mobile/test_lite_script_module.py TestLiteScriptModule

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27542502](https://our.internmc.facebook.com/intern/diff/D27542502)

[ghstack-poisoned]

raziel commented on torch/csrc/jit/serialization/export_module.cpp:
```diff
@@ -154,6 +137,25 @@ std::pair<IValue, IValue> getFunctionTuple(
+      }
+    }
+    // Note 1-to-1 correspondence between instructions and debug handles
+    op_debug_handles.emplace_back(debug_handle);
```
Ok, so now there's no more room for debug_handle_manager not being available. Thanks!

kimishpatel committed:
Update on "[Pytorch Mobile DebugInfo Serialization] Save debug handles for all instructions."


Summary:
Earlier for bytecode serialization we were saving debug handles only for OPs and not all
instructions. This PR makes changes to add that for all instructions.

Test Plan:
python test/mobile/test_lite_script_module.py TestLiteScriptModule

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27542502](https://our.internmc.facebook.com/intern/diff/D27542502)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Pytorch Mobile DebugInfo Serialization] Save debug handles for all instructions."


Summary:
Earlier for bytecode serialization we were saving debug handles only for OPs and not all
instructions. This PR makes changes to add that for all instructions.

Test Plan:
python test/mobile/test_lite_script_module.py TestLiteScriptModule

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27542502](https://our.internmc.facebook.com/intern/diff/D27542502)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Pytorch Mobile DebugInfo Serialization] Save debug handles for all instructions."


Summary:
Earlier for bytecode serialization we were saving debug handles only for OPs and not all
instructions. This PR makes changes to add that for all instructions.

Test Plan:
python test/mobile/test_lite_script_module.py TestLiteScriptModule

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27542502](https://our.internmc.facebook.com/intern/diff/D27542502)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Pytorch Mobile DebugInfo Serialization] Save debug handles for all instructions."


Summary:
Earlier for bytecode serialization we were saving debug handles only for OPs and not all
instructions. This PR makes changes to add that for all instructions.

Test Plan:
python test/mobile/test_lite_script_module.py TestLiteScriptModule

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27542502](https://our.internmc.facebook.com/intern/diff/D27542502)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Pytorch Mobile DebugInfo Serialization] Save debug handles for all instructions."


Summary:
Earlier for bytecode serialization we were saving debug handles only for OPs and not all
instructions. This PR makes changes to add that for all instructions.

Test Plan:
python test/mobile/test_lite_script_module.py TestLiteScriptModule

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27542502](https://our.internmc.facebook.com/intern/diff/D27542502)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Pytorch Mobile DebugInfo Serialization] Save debug handles for all instructions."


Summary:
Earlier for bytecode serialization we were saving debug handles only for OPs and not all
instructions. This PR makes changes to add that for all instructions.

Test Plan:
python test/mobile/test_lite_script_module.py TestLiteScriptModule

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27542502](https://our.internmc.facebook.com/intern/diff/D27542502)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Pytorch Mobile DebugInfo Serialization] Save debug handles for all instructions."


Summary:
Earlier for bytecode serialization we were saving debug handles only for OPs and not all
instructions. This PR makes changes to add that for all instructions.

Test Plan:
python test/mobile/test_lite_script_module.py TestLiteScriptModule

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27542502](https://our.internmc.facebook.com/intern/diff/D27542502)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Pytorch Mobile DebugInfo Serialization] Save debug handles for all instructions."


Summary:
Earlier for bytecode serialization we were saving debug handles only for OPs and not all
instructions. This PR makes changes to add that for all instructions.

Test Plan:
python test/mobile/test_lite_script_module.py TestLiteScriptModule

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27542502](https://our.internmc.facebook.com/intern/diff/D27542502)

[ghstack-poisoned]

kimishpatel committed:
Update on "[Pytorch Mobile DebugInfo Serialization] Save debug handles for all instructions."


Summary:
Earlier for bytecode serialization we were saving debug handles only for OPs and not all
instructions. This PR makes changes to add that for all instructions.

Test Plan:
python test/mobile/test_lite_script_module.py TestLiteScriptModule

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D27542502](https://our.internmc.facebook.com/intern/diff/D27542502)

[ghstack-poisoned]
